File: square_digits.rb
def square_digits num
  num.to_s.chars.map { |digit| digit.to_i**2 }.join.to_i
end

puts square_digits(3212) # 9414
puts square_digits(2112) # 4114
puts square_digits(1111) # 1111
puts square_digits(1234321) # 14916941
puts square_digits(0) # 0

File: squareDigits.js
function squareDigits(num) {
  const numStr = num.toString().split("")
  let array = []
  for (let i = 0;i < numStr.length;i += 1){
    array.push((numStr[i]) **2)
  }
  return +array.join('');
}

console.log(squareDigits(3212)); //9414

console.log(squareDigits(2112)); //4114

console.log(squareDigits(0)); //0
